File: packages/python/examples/flow_mermaid.py
from axllm import flow


source = """flowchart TD
  %%ax classify: requestText:string -> route:class \"support, sales\"
  %%ax reply: requestText:string -> replyText:string(max 300)
  classify{route} -->|support| reply
"""

program = flow(source)
rendered = str(program)
assert "%%ax reply: requestText:string -> replyText:string(max 300)" in rendered
assert "classify -->|support| reply" in rendered
assert str(flow(rendered)) == rendered
print("python-flow-mermaid-ok")

File: packages/cpp/examples/flow_mermaid.cpp
#include "axllm/axllm.hpp"
#include <iostream>

int main() {
  const std::string source = R"(flowchart TD
  %%ax classify: requestText:string -> route:class "support, sales"
  %%ax reply: requestText:string -> replyText:string(max 300)
  classify{route} -->|support| reply)";
  auto program = axllm::flow(source);
  const std::string rendered = program.str();
  if (rendered.find("%%ax reply: requestText:string -> replyText:string(max 300)") == std::string::npos) return 1;
  if (rendered.find("classify -->|support| reply") == std::string::npos) return 2;
  if (axllm::flow(rendered).str() != rendered) return 3;
  std::cout << "cpp-flow-mermaid-ok\n";
}
